### PR TITLE
Add more options in dhcpd.conf

### PR DESCRIPTION
**Pull request by tampakrap**
- global options
- failover peer block
- more options (and pool block) in the subnet block

### CODE REVIEW

**Comment by gravyboat** [comment]
This change looks good, could you please add pillar examples for the new items? Once that's done I'll get this merged in.

**Comment by tampakrap** [comment]
Added, let me know if they are sufficient please

**Comment by tampakrap** [comment]
Don't merge it please, I found a bug, I will file an inline comment

File: dhcpd/files/subnet.jinja [hunk 1]
```diff
@@ -31,15 +31,36 @@
...
+{{ intendation }}  option tftp-server-name "{{ config['tftp_server_name'] }}";
+{%- endif %}
+{%- if config.has_key('smtp_server') %}
+{{ intendation }}  option smtp-server {{ config['smtp_server'] }};
+{%- endif %}
 {%- if config.has_key('domain_name') %}
 {{ intendation }}  option domain-name "{{ config['domain_name'] }}";
 {%- endif %}
+{%- if config.has_key('domain_search') %}
+{{ intendation }}  option domain-search "{{ config['domain_search']|join(',') }}";
```
Comment (tampakrap) [inline dhcpd/files/subnet.jinja]: this one is buggy: it creates entries like `"example1.com,example2.com,example3.com"` while we actually need `"example1.com","example2.com","example3.com"`. I am not sure how to do it though, suggestions welcome

Comment (tampakrap) [inline dhcpd/files/subnet.jinja]: fixed

**Comment by tampakrap** [comment]
My issue is fixed, feel free to proceed

**Comment by gravyboat** [comment]
Thanks @tampakrap! I'm merging this in and squashing the commits.

**Pull request merged**